src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -260,6 +260,57 @@ public TypeSymbolWithAnnotations SetIsAnnotated(CSharpCompilation compilation)
         public TypeSymbolWithAnnotations AsNullableReferenceType() => _extensions.AsNullableReferenceType(this);
         public TypeSymbolWithAnnotations AsNotNullableReferenceType() => _extensions.AsNotNullableReferenceType(this);
 
+        public bool TryMergeNullability(TypeSymbolWithAnnotations other, VarianceKind variance, out TypeSymbolWithAnnotations merged)
+        {
+            TypeSymbol type;
+            bool? isNullable;
+            if (TypeSymbol.TryMergeNullability(other.TypeSymbol, out type) &&
+                TryMergeIsNullable(IsNullable, other.IsNullable, variance, out isNullable))
+            {
+                merged = Create(type, isNullable, CustomModifiers);
+                return true;
+            }
+            merged = default;
+            return false;
+        }
+
+        private static bool TryMergeIsNullable(bool? a, bool? b, VarianceKind variance, out bool? merged)
+        {
+            if (a == b)
+            {
+                merged = a;
+            }
+            else
+            {
+                switch (variance)
+                {
+                    case VarianceKind.In:
+                        merged = (a == false || b == false) ? (bool?)false : null;
```

Comment:
>null [](start = 77, length = 4)

If both a and b are true then true should be the result here.

Reply:
`(a == b)` is handled earlier.

---
In reply to: [223453835](https://github.com/dotnet/roslyn/pull/29727#discussion_r223453835) [](ancestors = 223453835)